src/containers/flows/AcceptBid/AcceptBid.tsx

```diff
@@ -0,0 +1,647 @@
+	const [isCheckingAllowance, setIsCheckingAllowance] = useState(false);
+	const [isBidConfirmed, setIsBidConfirmed] = useState(false);
+	const [hasBidDataLoaded, setHasBidDataLoaded] = useState(false);
+	const [hasApprovedZAuction, setHasApprovedZAuction] = useState<
+		boolean | undefined
+	>();
+
+	///////////////
+	// Functions //
+	///////////////
+
+	// Steps Functions
+	const approveZAuction = async () => {
+		setError(NO_ERROR);
+		setStatusText(CHECK_GAS_STATUS);
+		setIsApprovalInProgress(true);
+
+		// Check User Balance (Gas)
+		try {
+			const gasRequired = await wildContract.estimateGas.approve(
+				zAuctionAddress,
+				ethers.constants.MaxUint256,
+			);
+			const gasPrice = await wildContract.signer.getGasPrice();
+			const userBalance = await wildContract.signer.getBalance();
+
+			const totalPrice = gasPrice.mul(gasRequired);
+			if (userBalance.lt(totalPrice)) {
+				setError(LOW_BALANCE);
+				setIsApprovalInProgress(false);
+				return;
+			}
+		} catch (e) {
+			console.error(e);
+			setError(FAIL_TO_CALCULATE_GAS);
+			setIsApprovalInProgress(false);
+			return;
+		}
+
+		// User Wallet Interaction
+		setStatusText(ACCEPT_ZAUCTION_PROMPT);
+
+		let tx: Maybe<ethers.ContractTransaction>;
+		try {
+			tx = await wildContract.approve(
+				zAuctionAddress,
+				ethers.constants.MaxUint256,
+			);
+		} catch (e) {
+			console.error(e);
+			if (e.code === 4001) {
+				setError(APPROVAL_REJECTED);
+			} else {
+				setError(FAILED_TRANSACTION);
+			}
+			setIsApprovalInProgress(false);
+			return;
+		}
+
+		// Await Approval
+		setStatusText(APPROVING_ZAUCTION);
+		try {
+			await tx.wait();
+		} catch (e) {
+			setError(FAILED_TRANSACTION);
+			console.error(e);
+			setIsApprovalInProgress(false);
+			return;
+		}
+
+		// Complete zAuction Approval
+		setHasApprovedZAuction(true);
+		setIsApprovalInProgress(false);
+	};
+
+	const acceptBidConfirmed = async () => {
+		setError(NO_ERROR);
+		if (!acceptingBid) {
+			return;
+		}
+		setIsAccepting(true);
+		setStatusText(AWAITING_APPROVAL);
+
+		// User Wallet Interaction
+		let tx: Maybe<ethers.ContractTransaction>;
+		try {
+			tx = await acceptBid(acceptingBid.bid);
+			setStatusText(ACCEPTING_BID);
+			// Set Accept Step
+			setStep(Steps.Accept);
+			if (tx) {
+				await tx.wait();
+			}
+			setTimeout(() => {
+				ownedQuery.refetch();
+			}, 500);
+			setIsBidConfirmed(true);
+		} catch (e) {
+			setError(APPROVAL_REJECTED);
+			setIsAccepting(false);
+		}
+		try {
+			await tx?.wait();
+		} catch (e) {
+			setError(FAILED_TRANSACTION);
+			console.error(e);
+			setIsAccepting(false);
+		}
+		setIsAccepting(false);
+	};
+
+	const checkZAuctionApproval = async () => {
+		setIsCheckingAllowance(true);
+		setStatusText(CHECKING_ZAUCTION_STATUS);
+		await new Promise((r) => setTimeout(r, 800));
+		if (hasApprovedZAuction) {
+			setIsCheckingAllowance(false);
+			setStep(Steps.Confirm);
+		} else {
+			setIsCheckingAllowance(false);
+			setHasApprovedZAuction(false);
+		}
+	};
```

Comment:
Hey @colbr, I've reverted the changes here. When applying ```back``` to the arrow link this seems to just spin the arrow 180° rather than place the arrow to the left of the text. Not sure if I'm missing something there but I've tweaked this so if ```back``` is applied, it will place the arrow to the left of the text. Seems to work ok here and in other instances its used but let me know if this needs changing.

Reply:
Updated mate 👍  I'm not entirely sure whats happening under the hood here though. Is it```allowance``` that stores whether the userId has approved the zAuction?

```
	const checkZAuctionApproval = async () => {
		const allowance = await wildContract.allowance(userId!, zAuctionAddress);
		const bidAsWei = ethers.utils
			.parseEther(String(acceptingBid.bid.amount))
			.toString();
		const needsApproving = allowance.lt(bidAsWei);
		setIsCheckingAllowance(true);
		setStatusText(CHECKING_ZAUCTION_STATUS);
		await new Promise((r) => setTimeout(r, 800));
		if (hasApprovedZAuction) {
			setIsCheckingAllowance(false);
			setStep(Steps.Confirm);
		} else {
			setIsCheckingAllowance(false);
			setHasApprovedZAuction(!needsApproving);
		}
	};
```

Reply:
I believe the psuedocode of the above is:
```
get (from the WILD contract) the amount of WILD the user has approved zAuction to spend
convert the bid amount to wei
compare bid amount to approved amount
if approved amount is more than or equal to bid amount, continue
```

The approval (or at least the web3 side of it) will be done in another task so dw about it too much

Reply:
nice one Brett thanks for that breakdown 👍